src/mesh/X3D_Draco_Mesh_Reader.hh

```diff
@@ -0,0 +1,133 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   mesh/X3D_Draco_Mesh_Reader.cc
+ * \author Ryan Wollaeger <wollaeger@lanl.gov>, Kendra Keady
+ * \date   Wednesday, Jul 11, 2018, 14:24 pm
+ * \brief  X3D_Draco_Mesh_Reader header file.
+ * \note   Copyright (C) 2018 Los Alamos National Security, LLC.
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#ifndef rtt_mesh_X3D_Draco_Mesh_Reader_hh
+#define rtt_mesh_X3D_Draco_Mesh_Reader_hh
+
+#include "Draco_Mesh_Reader.hh"
+#include "ds++/Assert.hh"
+#include "ds++/config.h"
+#include <map>
+#include <string>
+#include <vector>
+
+namespace rtt_mesh {
+
+//===========================================================================//
+/*!
+ * \class X3D_Draco_Mesh_Reader
+ *
+ * \brief Read x3d file format and supply data to Draco_Mesh_Builder.
+ *
+ * This class parses mesh data from an x3d file format and manipulates it into a
+ * form that is compatible to the Draco_Mesh_Builder class.  The parsing of the
+ * file stream to the map follows work by Kendra Keady on a parsing framework.
+ *
+ * \todo: Boundary condition data is evidently parsed separately in X3D, so
+ * meshes generated from this reader will not have side flag data (which ids
+ * boundary conditions.
+ *
+ * \todo: Consider using the Class_Parse_Table formalism developed by Kent Budge
+ * as an alternative.
+ */
+//===========================================================================//
+
+class X3D_Draco_Mesh_Reader : public Draco_Mesh_Reader {
+public:
+  // >>> TYPEDEFS
+  typedef std::pair<std::string, std::vector<std::string>> Parsed_Element;
+  typedef std::vector<Parsed_Element> Parsed_Elements;
+
+private:
+  // >>> DATA
+
+  //! File name
+  std::string filename;
+
+  //! Vector of all parsed key-value data pairs (includes valueless delimiters)
+  Parsed_Elements parsed_pairs;
+
+  //! Header data map (header in x3d file)
+  std::map<std::string, std::vector<int>> x3d_header_map;
+
+  //! Node coordinates
+  std::map<int, std::vector<double>> x3d_coord_map;
+
+  //! Face-to-node map
+  std::map<int, std::vector<int>> x3d_facenode_map;
+
+  //! Cell-to-face map
+  std::map<int, std::vector<int>> x3d_cellface_map;
+
+public:
+  //! Constructor
+  explicit X3D_Draco_Mesh_Reader(std::string filename_);
+
+  // >>> SERVICES
+
+  DLL_PUBLIC_mesh void read_mesh();
+
+  // >>> ACCESSORS
+
+  // header data
+  unsigned get_process() const { return x3d_header_map.at("process")[0] - 1; }
+  unsigned get_numdim() const { return x3d_header_map.at("numdim")[0]; }
+  unsigned get_numcells() const { return x3d_header_map.at("elements")[0]; }
+  unsigned get_numnodes() const { return x3d_header_map.at("nodes")[0]; }
+
```

Comment:
Again, how difficult is it to use `int64_t` for `numcell`, `numnodes`?  Maybe we need a new ticket to review the 2 billion cell issue.

Reply:
An attempt is underway on this - thank you for the input.

Reply:
Delay adding uint64_t interfaces.